File: 알고리즘 문제풀이/프로그래머스/단속카메라.py

"""
routes을 sort한 뒤,그리디 방법으로 범위를 탐색하면서 코딩을 진행
"""
def solution(routes):
    routes.sort()
    tmp,answer=routes[0][1],1
    for i in range(1,len(routes)):
        if tmp>routes[i][1]:
            tmp=routes[i][1]
        elif tmp<routes[i][0]: #범위를 넘는 값에서 다음 차가 출발 했다면 카메라 증가
            tmp=routes[i][1]
            answer+=1
    return answer

File: 알고리즘 문제풀이/프로그래머스/단속카메라.js
function solution(routes) {
  let answer = 1,
    ri = 0;
  routes.sort((x, y) => (x[0] < y[0] ? -1 : 1));
  ri = routes[0][1];
  for (let i = 1; i < routes.length; i++) {
    if (routes[i][1] < ri) ri = routes[i][1];
    else if (routes[i][0] > ri) {
      answer += 1;
      ri = routes[i][1];
    }
  }
  return answer;
}
